android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/RestoreWalletFragment.java

```diff
@@ -65,38 +65,48 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
             }
         });
 
-        EditText passwordEdittext = view.findViewById(R.id.restore_wallet_password);
-        EditText retypePasswordEdittext = view.findViewById(R.id.restore_wallet_retype_password);
-
         Button secureCryptoButton = view.findViewById(R.id.btn_restore_wallet);
         secureCryptoButton.setOnClickListener(v -> {
-            String passwordInput = passwordEdittext.getText().toString().trim();
-            String retypePasswordInput = retypePasswordEdittext.getText().toString().trim();
-
-            if (passwordInput.isEmpty()
-                    || !Utils.PASSWORD_PATTERN.matcher(passwordInput).matches()) {
-                passwordEdittext.setError(getResources().getString(R.string.password_text));
-            } else if (retypePasswordInput.isEmpty()
-                    || !passwordInput.equals(retypePasswordInput)) {
-                retypePasswordEdittext.setError(
-                        getResources().getString(R.string.retype_password_error));
-            } else {
-                KeyringController keyringController = getKeyringController();
-                if (keyringController != null) {
-                    keyringController.restoreWallet(recoveryPhraseText.getText().toString().trim(),
-                            passwordEdittext.getText().toString().trim(), false, result -> {
-                                if (result) {
-                                    Utils.hideKeyboard(getActivity());
-                                    onNextPage.gotoNextPage(true);
-                                    Utils.setCryptoOnboarding(false);
-                                } else {
-                                    Toast.makeText(getActivity(), R.string.account_recovery_failed,
-                                                 Toast.LENGTH_SHORT)
-                                            .show();
-                                }
-                            });
+            EditText passwordEdittext = view.findViewById(R.id.restore_wallet_password);
+            String passwordInput = passwordEdittext.getText().toString();
+
+            KeyringController keyringController = getKeyringController();
+            assert keyringController != null;
```

Comment:
Why it is `assert` here? Won't that cause a crash if `KeyringController` is not ready for some reason?

Reply:
it shouldn't happen. That fragment is called from `BraveWalletActivity`. All native stuff is loaded there. Everything should be up when we call that fragment.

Reply:
plus the keyringcontroller access is inside setOnClickListener

Reply:
Ah ok that makes sense.